From e6cd17976621a2ee2d0ed8e1ffbf9e4bdcafb657 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 10 Feb 2020 14:32:42 +0200
Subject: [PATCH] [SW-0] goya2: fix loop condition on fw min version

looping on lower dword and LSB are missing in both
external and internal loop in function
security_find_min_fw_version()

Change-Id: I709790f74a1b3d891de3dfdeed81080574e7d8dc
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index f30bd66..ba91579 100644
--- a/common/security.c
+++ b/common/security.c
@@ -426,15 +426,15 @@
  */
 uint8_t security_find_min_fw_version(uint32_t *fw_version)
 {
-	uint8_t row, colume;
+	int8_t row, colume;
 	uint8_t fw_version_sec;
-
 	/* init max version */
 	fw_version_sec = SECURITY_MAX_FW_SEC_VERSION;
 
 	/* iterate from end to start, till a set bit is found */
-	for (row = SECURITY_FW_VER_LAST_DWORD; row > 0; row--) {
-		for (colume = SECURITY_FW_VER_LAST_BYTE; colume > 0; colume--) {
+	for (row = SECURITY_FW_VER_LAST_DWORD; row >= 0; row--) {
+		for (colume = SECURITY_FW_VER_LAST_BYTE; colume >= 0;
+				colume--) {
 			/* find highest bit set in valid FW */
 			fw_version_sec--;
 			if (fw_version[row] & BIT(colume))
